File: engine/model/src/main/scala/client/inventory/ModifyInventory.scala
package client.inventory

class ModifyInventory(private var _mode: Int, private var _item: Item, private var _oldPos: Short) {
  def mode: Int = _mode

  def item: Item = _item

  def oldPos: Short = _oldPos

  def this(mode: Int, item: Item) = {
    this(mode, item, 0)
  }

  def clear(): Unit = {
    _item = null
  }

  def position(): Short = {
    item.position
  }

  def inventoryType(): Int = {
    item.inventoryType.getType
  }

  def quantity(): Short = {
    item.quantity
  }
}

File: engine/engine-model/src/main/java/client/inventory/ModifyInventory.java
package client.inventory;

public record ModifyInventory(int mode, Item item, short oldPos) {
   public ModifyInventory(int mode, Item item) {
      this(mode, item, (short) 0);
   }

   public ModifyInventory clear() {
      return new ModifyInventory(mode, null, oldPos);
   }

   public short position() {
      return item.position();
   }

   public int inventoryType() {
      return item.inventoryType().getType();
   }

   public short quantity() {
      return item.quantity();
   }
}
